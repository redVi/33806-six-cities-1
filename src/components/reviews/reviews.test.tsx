import React from "react";
import renderer from "react-test-renderer";
import Reviews from "@/components/reviews/reviews";

const comment = {
  id: 1,
  rating: 4.2,
  comment: "Nice place",
  date: "2019-06-10T02:34:08.739Z",
  user: {
    name: "Alex",
    avatarUrl: "alex-avatar.jpg"
  }
};

describe("Reviews", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Reviews id={50} isLoggedIn={false} comments={[comment]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
