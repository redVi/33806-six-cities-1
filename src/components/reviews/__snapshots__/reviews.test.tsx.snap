// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews renders correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="alex-avatar.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Alex
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "84%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Nice place
        </p>
        <time
          className="reviews__time"
          dateTime="2019-06-10T02:34:08.739Z"
        >
          June 2019
        </time>
      </div>
    </li>
  </ul>
</section>
`;
